haystack/nodes/preprocessor/preprocessor.py

```diff
@@ -444,3 +450,49 @@ def _find_longest_common_ngram(
             # no common sequence found
             longest = ""
         return longest if longest.strip() else None
+
+    def _sentence_tokenizer(self, text: str) -> List[str]:
+        """
+        Tokenize text into sentences.
+        :param text: str, text to tokenize
+        :param language: str, language of text
+        :param tokenizer_model_path: Path, path to tokenizer model
+        :return: list[str], list of sentences
+        """
+        sentences = []
+        language_name = iso639_to_nltk.get(self.language)
+        tokenizer_model_path = (
+            Path(self.tokenizer_model_folder).absolute() / f"{self.language}.pickle"
+            if self.tokenizer_model_folder is not None
+            else None
+        )
+        if tokenizer_model_path is not None:
+            try:
+                sentence_tokenizer = nltk.data.load(f"file:{str(tokenizer_model_path)}", format="pickle")
+                sentences = sentence_tokenizer.tokenize(text)
+            except LookupError:
+                logger.warning(f"PreProcessor couldn't load sentence tokenizer from {str(tokenizer_model_path)}")
+            except (UnpicklingError, ValueError) as e:
+                logger.warning(
+                    f"PreProcessor couldn't determine model format of sentence tokenizer at {str(tokenizer_model_path)}. - Exception: {str(e)}"
+                )
+            if not sentences and not language_name:
+                logger.warning(
+                    f"PreProcessor couldn't find default or custom sentence tokenizer model for {self.language}. Using English instead."
+                )
+                sentences = nltk.tokenize.sent_tokenize(text, language="english")
+            elif not sentences and language_name:
+                logger.warning(
+                    f"PreProcessor couldn't find custom sentence tokenizer model for {self.language}. Using default {self.language} model."
+                )
+                sentences = nltk.tokenize.sent_tokenize(text, language=language_name)
+        elif not language_name:
+            logger.warning(
+                f"PreProcessor couldn't find default sentence tokenizer model for {self.language}. Using English instead. \
+                    You may train your own model and use the tokenizer_model_folder parameter."
+            )
+            sentences = nltk.tokenize.sent_tokenize(text, language="english")
+        else:
+            sentences = nltk.tokenize.sent_tokenize(text, language=language_name)
+
```

Comment:
Here is my take on the same code. By using many `return` I reduce the number of `if-else`. I added a few comments and note that I raised all log messages to `ERROR`: I think they're quite important, let's make the user aware.

There should be no difference with your version. No need to choose between these two, feel free to mix them :slightly_smiling_face: 

```suggestion

        language_name = iso639_to_nltk.get(self.language)
        
        # Try to load a custom model from 'tokenizer_model_path'
        if self.tokenizer_model_folder:
            tokenizer_model_path = Path(self.tokenizer_model_folder).absolute() / f"{self.language}.pickle"
            try:
                sentence_tokenizer = nltk.data.load(f"file:{str(tokenizer_model_path)}", format="pickle")
                sentences = sentence_tokenizer.tokenize(text)
            except LookupError:
                logger.exception(f"PreProcessor couldn't load sentence tokenizer from {str(tokenizer_model_path)}")
            except (UnpicklingError, ValueError) as e:
                logger.exception(
                    f"PreProcessor couldn't determine model format of sentence tokenizer at {str(tokenizer_model_path)}."
                )
            if sentences:
                return sentences
            
            # NLTK failed to split, fallback to the default model or to English
            if language_name:
               logger.error(
                    f"PreProcessor couldn't find custom sentence tokenizer model for {self.language}. Using default {self.language} model."
                )
                return nltk.tokenize.sent_tokenize(text, language=language_name)
             logger.error(
                f"PreProcessor couldn't find default or custom sentence tokenizer model for {self.language}. Using English instead."
            )
            return nltk.tokenize.sent_tokenize(text, language="english")
        
        # Use a default NLTK model
        if language_name:
            return nltk.tokenize.sent_tokenize(text, language=language_name)
        logger.error(
            f"PreProcessor couldn't find default sentence tokenizer model for {self.language}. Using English instead. "
            "You may train your own model and use the 'tokenizer_model_folder' parameter."
        )
        return nltk.tokenize.sent_tokenize(text, language="english")
```

Reply:
:smiling_face_with_tear: What a shame! 

When I did the flag tries, there wasn't this function. Code was inside another method, so the returns were not possible. After the method creation I didn't took a look and kept that in mind lol.

This is what I can call a silly mistake.